mllib/src/test/scala/org/apache/spark/ml/feature/QuantileDiscretizerSuite.scala

```diff
@@ -83,10 +83,20 @@ class QuantileDiscretizerSuite
       .setOutputCol("result")
       .setNumBuckets(numBuckets)
 
-    // Reserve extra one bucket for NaN
-    val expectedNumBuckets = discretizer.fit(df).getSplits.length - 1
-    val result = discretizer.fit(df).transform(df)
-    val observedNumBuckets = result.select("result").distinct.count
+    // Reserve extra one bucket for NaN values
+    discretizer.setHandleInvalid("keep")
+    var expectedNumBuckets = discretizer.fit(df).getSplits.length - 1
```

Comment:
I'd also recommend directly testing the values of the splits.  The current test makes sure that handleInvalid is passed to the bucketizer correctly, which is important but separate.

Also, please use vals (not vars) for clarity.  I'd recommend making a helper method for lines 87-93, which can be reused for the test of handleInvalid = "skip"